### PR TITLE
`Development`: Remove string concatenation in logger calls

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).

### Motivation and Context
When logging messages with parameters, the message should not be built using string concatenation. This allows for better performance since this new string is now only built when actually needed.

A PR like this was already created before, see #3122. I tried adding this to Codacy, but appartently it's not possible to detect that using our current configuration / without adding completly new tools.

This PR also mentioned "There are a few cases (about 30) remaining where a string is constructed via +. This is due to the fact that when an exception should be passed on to the logger to print, the additional parameters are not allowed." This is no longer true, see https://stackoverflow.com/a/6374166

### Description
Instead of concatenating the arguments they now get passed as additional parameters.

### Steps for Testing
Code Review

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabUserManagementService.java [hunk 1]
```diff
@@ -137,7 +137,7 @@ public void updateCoursePermissions(Course updatedCourse, String oldInstructorGr
         }
 
         final List<ProgrammingExercise> programmingExercises = programmingExerciseRepository.findAllProgrammingExercisesInCourseOrInExamsOfCourse(updatedCourse);
-        log.info("Update Gitlab permissions for programming exercises: " + programmingExercises.stream().map(ProgrammingExercise::getProjectKey).toList());
+        log.info("Update Gitlab permissions for programming exercises: {}", programmingExercises.stream().map(ProgrammingExercise::getProjectKey).toList());
```
Comment (Marvin-Vitcu) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabUserManagementService.java:140]: in this entire class: Gitlab ---> GitLab

File: src/main/java/de/tum/in/www1/artemis/service/user/UserService.java [hunk 2]
```diff
@@ -285,7 +285,7 @@ public User registerUser(UserDTO userDTO, String password) {
                 vcsUserManagementService.deactivateUser(savedNonActivatedUser.getLogin());
             }
             catch (VersionControlException e) {
-                log.error("An error occurred while registering GitLab user " + savedNonActivatedUser.getLogin() + ":", e);
+                log.error("An error occurred while registering GitLab user {}:", savedNonActivatedUser.getLogin(), e);
```
Comment (Marvin-Vitcu) [inline src/main/java/de/tum/in/www1/artemis/service/user/UserService.java:288]:
```suggestion
                log.error("An error occurred while registering GitLab user {}: ", savedNonActivatedUser.getLogin(), e);
```

File: src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabUserManagementService.java (hunk 1 above)
Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabUserManagementService.java:140]: Gitlab is used in many places, a change like this is out of scope of this PR.
grafik

File: src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java [hunk 3]
```diff
@@ -293,8 +293,9 @@ public ResponseEntity<Void> testCaseChanged(@PathVariable Long exerciseId, @Requ
             log.info("create new programmingSubmission with commitHash: {} for exercise {}", lastCommitHash, exerciseId);
         }
         catch (Exception ex) {
-            log.debug("Commit hash could not be parsed for from test repository from exercise " + exerciseId
-                    + ", the submission will be created with the latest commitHash of the solution repository.", ex);
+            log.debug(
+                    "Commit hash could not be parsed for from test repository from exercise {}, the submission will be created with the latest commitHash of the solution repository.",
```
Comment (Marvin-Vitcu) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingSubmissionResource.java:297]:
```suggestion
                    "Commit hash could not be parsed from test repository from exercise {}, the submission will be created with the latest commit hash of the solution repository.",
```

File: src/main/java/de/tum/in/www1/artemis/service/user/UserService.java (hunk 2 above)
Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/service/user/UserService.java:288]: I don't see a need for this, in fact I think that message should not end with a space.

Comment (Marvin-Vitcu) [reply src/main/java/de/tum/in/www1/artemis/service/user/UserService.java:288]: I wasn't really sure for this one, as some log.error ended with a space some didn't

**Review by Marvin-Vitcu — DISMISSED** [review]

File: src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabUserManagementService.java (hunk 1 above)
Comment (Strohgelaender) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/gitlab/GitLabUserManagementService.java:140]: But this is out of scope for this PR, we could consider it in a follow-up.

**Review by b-fein — APPROVED** [review]
:+1: 

When using the SpotBugs plugin in the local IDE it recommends using the proper log formatting. I’m also not aware that one of the CI linters we use could be configured to check for it.

**Review by yuling-sun — APPROVED** [review]
Reviewed changes. LGTM.

**Review by bassner — APPROVED** [review]

**Pull request merged**